eureka/S3_data_reduction/niriss.py

```diff
@@ -1,2 +1,435 @@
+    
+
+    for i in range(summed.shape[1]):
+
+        # Identifies peaks in the F277W filtered image
+        fp = identify_peaks(summed_f277[:,i], height=100000, distance=10)
+        if len(fp)==2:
+            f277_peaks[i] = fp
+    
+        if i < double_peaked[0]:
+            height=2000
+        elif i >= double_peaked[0] and i < double_peaked[1]:
+            height = 100
+        elif i >= double_peaked[1]:
+            height = 5000
+            
+        p = identify_peaks(new_ccd_no_premask[:,i].data, height=height, distance=10)
+        if i < 900:
+            p = p[p>40] # sometimes catches an upper edge that doesn't exist
+        
+        peaks[i][:len(p)] = p
+
+    # Removes 0s from the F277W boundaries
+    xf = np.arange(0,summed_f277.shape[1],1)
+    good = f277_peaks[:,0]!=0
+    xf=xf[good]
+    f277_peaks=f277_peaks[good]
+
+    # Fitting a polynomial to the boundary of each order
+    x = np.arange(0,new_ccd_no_premask.shape[1],1)
+    avg = np.zeros((new_ccd_no_premask.shape[1], 6))
+
+    for ind in range(4): # CHANGE THIS TO 6 TO ADD THE THIRD ORDER
+        q = peaks[:,ind] > 0
+        
+        # removes outliers
+        diff = np.diff(peaks[:,ind][q])
+        good = np.where(np.abs(diff)<=np.nanmedian(diff)+2*np.nanstd(diff))
+        good = good[5:-5]
+        y = peaks[:,ind][q][good] + 0
+        y = y[x[q][good]>xf[-1]]
+        
+        # removes some of the F277W points to better fit the 2nd order
+        if ind < 2:
+            cutoff=-1
+        else:
+            cutoff=250
+
+        xtot = np.append(xf[:cutoff], x[q][good][x[q][good]>xf[-1]])
+        if ind == 0 or ind == 2:
+            ytot = np.append(f277_peaks[:,0][:cutoff], y)
+        else:
+            ytot = np.append(f277_peaks[:,1][:cutoff], y)
+        
+        # Fits a 4th degree polynomiall
+        poly= np.polyfit(xtot, ytot, deg=4)
+        fit = np.poly1d(poly)
+            
+        avg[:,ind] = fit(x)
+
+    if isplots:
+        plt.imshow(summed, vmin=0, vmax=2e3)
+        plt.plot(x, np.nanmedian(avg[:,:2],axis=1), 'k', lw=2)
+        plt.plot(x, np.nanmedian(avg[:,2:4],axis=1), 'r', lw=2)
+        plt.show()
+    
+
+    tab = Table()
+    tab['x'] = x
+    tab['order_1'] = np.nanmedian(avg[:,:2],axis=1)
+    tab['order_2'] = np.nanmedian(avg[:,2:4],axis=1)
+
+    if save:
+        tab.write('niriss_order_fits_method2.csv',format='csv')
+
+    return tab
+
+
+def simplify_niriss_img(data, meta, isplots=False):
+    """
+    Creates an image to map out where the orders are in
+    the NIRISS data.
+
+    Parameters     
+    ----------     
+    data : object  
+    meta : object 
+    isplots : bool, optional
+       An option to plot the data and intermediate steps to
+       retrieve the mask per each order. Default is False.  
+    """
+    perc  = np.nanmax(data.data, axis=0)
+
+    # creates data img mask
+    z,g = image_filtering(perc)
+    
+    if isplots:
+        plt.close()
```

Comment:
this close statement confuses me

Reply:
I think I was having trouble with this overplotting in my notebook with another figure. But by itself it works fine. I've removed the close statement and updated the isplots like the other ones (including a isplots>=5)